src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java

```diff
@@ -134,29 +127,9 @@ public static void failure(String k, Throwable error) {
         State s = State.get();
         StepContext c;
         synchronized (s) {
-            if (!s.contexts.containsKey(k)) {
-                LOGGER.info(() -> "Planning to unblock " + k + " as failure");
-                s.errors.put(k, error);
-                return;
-            }
-            c = getContext(s, k);
+            LOGGER.info(() -> "Planning to unblock " + k + " as failure");
+            s.errors.put(k, error);
         }
-        LOGGER.info(() -> "Unblocking " + k + " as failure");
-        c.onFailure(error);
-    }
-
-    /** @deprecated should not be needed */
-    @Deprecated
-    public StepContext getContext() {
-        State s = State.get();
-        synchronized (s) {
-            return getContext(s, k());
-        }
-    }
```

Comment:
Could be restored if needed, but I didn't see any callers in a quick search, and IDK how it would even be used in the first place. Deleting this though allows us to totally drop `State.contexts.`